cuqi/experimental/mcmc/_rto.py

```diff
@@ -197,10 +204,13 @@ def __init__(self, target=None, initial_point=None, maxit=100, stepsize="automat
         self.abstol = abstol   
         self.adaptive = adaptive
         self.maxit = maxit
+        self.inner_max_it = inner_max_it
+        self.penalty_parameter = penalty_parameter
 
     def _initialize(self):
         super()._initialize()
-        self._stepsize = self._choose_stepsize()
+        if callable(self.proximal):
```

Comment:
This logic is possibly difficult to understand in the future. I suggest you add an attribute like "_inner_solver" that can be either ADMM or FISTA (can just be a string or likewise). Then use checks like that to see which inner solver is used. Define the inner solver when the target is set.

Same holds for RegularizedGausssian

Reply:
I have now added the _inner_solver attribute.

I do not understand what you mean with 
> Same holds for RegularizedGausssian

Reply:
Thanks. Looks good. In both classes.